homeassistant/components/broadlink/device.py

```diff
@@ -24,41 +31,42 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def get_domains(device_type):
+def get_domains(device_type: str) -> set[Platform]:
     """Return the domains available for a device type."""
     return {d for d, t in DOMAINS_AND_TYPES.items() if device_type in t}
 
 
 class BroadlinkDevice:
     """Manages a Broadlink device."""
 
-    def __init__(self, hass, config):
+    api: blk.Device
+
+    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
         """Initialize the device."""
         self.hass = hass
         self.config = config
-        self.api = None
         self.update_manager = None
-        self.fw_version = None
-        self.authorized = None
-        self.reset_jobs = []
+        self.fw_version: int | None = None
+        self.authorized: bool | None = None
+        self.reset_jobs: list[CALLBACK_TYPE] = []
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the device."""
         return self.config.title
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str | None:
         """Return the unique id of the device."""
         return self.config.unique_id
 
     @property
-    def mac_address(self):
+    def mac_address(self) -> str:
         """Return the mac address of the device."""
         return self.config.data[CONF_MAC]
 
     @property
-    def available(self):
+    def available(self) -> bool | None:
```

Comment:
When does it return None?

Reply:
`self.update_manager.available` is initialised as `None`.

Reply:
We are casting None to False in the lines below

Reply:
That's a code-change, which I normally try to avoid.
But since it was requested by a code-owner I have adjusted accordingly.

Reply:
It's ok to leave as None for now, I can change another day after some tests